lib/ledger_sync/adaptors/quickbooks_online/searcher.rb

```diff
@@ -14,6 +14,35 @@ def previous_searcher
           paginate(limit: limit, offset: offset - limit)
         end
 
+        def query_string
+          ''
+        end
+
+        def resources
+          resource_class = self.class.inferred_resource_class
+
+          response = adaptor.query(
+            limit: limit,
+            offset: offset,
+            query: query_string,
+            resource_class: resource_class
+          )
+          return [] if response.body.blank?
+
+          response.body.dig(
+            'QueryResponse',
+            adaptor.class.ledger_resource_type_for(
+              resource_class: resource_class
+            ).classify
+          ).map do |c|
```

Comment:
@ryanwjackson All looks good to me except for one last thing. This `response.body.dig` can return `nil` which throws an error when we try to call `.map` on it. We could do something like this to set the default to an empty array

```
(response.body.dig(
   'QueryResponse',
    adaptor.class.ledger_resource_type_for(
      resource_class: resource_class
    ).classify
) || []).map do |c|
```

Reply:
Done!